app/controllers/FinalTaxCalculationController.scala

```diff
@@ -108,17 +99,22 @@ class FinalTaxCalculationController @Inject()(implicit val cc: MessagesControlle
 
   }
 
-  def agentSubmit(taxYear: Int): Action[AnyContent] = Authenticated.async { implicit request =>
-    implicit agent =>
-      getMtdItUserWithIncomeSources(incomeSourceDetailsService).flatMap { user =>
-        val fullName = user.session.get(clientFirstName).getOrElse("") + " " + user.session.get(clientLastName).getOrElse("")
-        agentFinalDeclarationSubmit(taxYear, fullName)(user, hc)
-      }
-  }
+//  def agentSubmit(taxYear: Int): Action[AnyContent] = auth.authenticatedAction(isAgent = true) { user =>
```

Comment:
Using auth.authenticatedAction here causes a unit test to fail in FinalTaxCalculationControllerSpec